### PR TITLE
CY-2598 - Fix react/destructuring-assignment warnings

### PR DESCRIPTION
**Pull request by qooban**
Reduced number of ESLint issues from:
✖ 7018 problems (0 errors, 7018 warnings)
to:
✖ 3702 problems (0 errors, 3702 warnings)

### Changes
- Most of the react/destructuring-assignment warnings fixed with codemod
- The rest react/destructuring-assignment issues fixed manually
- Fixed some security issues detected by NPM audit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "better-console": "^0.2.4",
     "blueimp-md5": "^2.7.0",
     "cloudify-blueprint-topology": "^2.3.0",
-    "cloudify-ui-common": "1.6.0",
```
Comment (kubama) [inline package.json:41]: Is this version upgrade (and others done in this file) somehow related to the topic of this PR?

File: app/components/AddWidgetModal.js [hunk 2]
```diff
@@ -116,22 +120,27 @@ export default class AddWidgetModal extends Component {
...
+        const { widgetsToAdd } = this.state;
+        return _.without(widgetsToAdd, widgetId);
     }
 
     uninstallWidget() {
-        const widgetId = this.state.widget.id;
+        const { onWidgetUninstalled } = this.props;
+        const {
+            widget: { id: widgetId }
+        } = this.state;
```
Comment (kubama) [inline app/components/AddWidgetModal.js:131]: This is a general comment, just using this code fragment for reference.
I understand the benefit of using destructuring assignments, but for nested cases, or for cases where const name is changed over the property name I'm in doubt. Looking at this case it seems neither easy to read nor intuitive. Personally I would stick with `const { widget } = this.state;` and then reference `widget.id`. What do you think?

File: package.json (hunk 1 above)
Comment (qooban) [reply package.json:41]: 1. cloudify-ui-common - this was done 'by the way' as 1.6.1 and 1.6.2 introduced some ESLint changes (see: https://github.com/cloudify-cosmo/cloudify-ui-common/releases/tag/v1.6.1 and https://github.com/cloudify-cosmo/cloudify-ui-common/releases/tag/v1.6.2)
2. the rest devDependencies - upgraded after doing `npm audit`

So the first one is related somehow. The second one is not related, but I'm not sure it is worth to divide it. If you think so, I can create separate PR.

File: app/components/AddWidgetModal.js (hunk 2 above)
Comment (qooban) [reply app/components/AddWidgetModal.js:131]: Yes, make sense. I'll go over such cases and try to simplify it.

**Pull request merged**